adrs/5541831634/README.md

```diff
@@ -0,0 +1,158 @@
+* You **must** use external service brand name as a subdirectory in
+  `<fractalDir>/services/[service name]`.
+* You **must** use `<fractalDir>/components` to store reusable React
+  components.
+* You **must** use `<fractalDir>/hooks` to store reusable React
+  hooks.
+* You **must** use `<fractalDir>/types` to store reusable TypeScript types.
+* You **must** use `<fractalDir>/styles` to store components styles such as
+  CSS or Stylesheets.
+* You **must** use `<fractalDir>/constants` to store constants variables.
+* You **must** use `<fractalDir>/contexts` to store React contexts definitions.
+* You **must** use `<fractalDir>/i18n` to store anything
+  related to Internationalization.
+* You **must** use `<fractalDir>/i18n/datetime` to store i18n date
+  and time formatting.
+* You **must** use `<fractalDir>/i18n/list` to store i18n sensitive
+  list formatting. Read more about the intent at [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat).
+* You **must** use `<fractalDir>/i18n/numbers` to store i18n number
+  formatting. Read more about the intent at [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat).
+* You **must** use `<fractalDir>/i18n/plural` to store i18n
+  pluralization formatting. Read more about the intent at [Intl.PluralRules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules).
+* You **must** use `<fractalDir>/i18n/relative-time` to store i18n
+  relative-time formatting. Read more about the intent at [Intl.RelativeTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat).
+* You **must** use `<fractalDir>/helpers/datetime` to store anything related
+  to Dates or Times. Except formatters, use `<fractalDir>/i18n/datetime`
+  instead.
+* You **must** use `<fractalDir>/helpers/currency` to store anything related
+  to currency. Except formatters, use `<fractalDir>/i18n/numbers`
+  instead.
+* You **must** use `<fractalDir>/helpers` to store anything that does not
+  fall into any of the other helper directory categories. You **may** create
+  your own files and directories based on your context.
+* You **must** use `<fractalDir>/transports/http` to store anything
+  related to HTTP transport clients.
+* You **must** use `<fractalDir>/transports/web-socket` to store
+  anything related to Web Socket transport clients.
+* You **must** use `<fractalDir>/transports/graphql` to store
+  anything related to GraphQL transport clients.
+* You **must** use `<fractalDir>/storages` to store anything related to the
+  device. You **may** use `local`, `session`, `cookies`, `websql`, and
+  `indexeddb` as subdirectories.
+* You **must** use `<fractalDir>/schemas` to store anything related to Joi, Yup
+  schemas or any similar package. You **may** use **joi**, **yup** or
+  package name as subdirectories.
```

Comment:
The last thing that is on my mind with this: if the goal is to reduce cognitive load, this is too many rules. I'm just imagining having to consult a rule book every time I want to create a new file. 

I think having some sort of tooling to enforce these rules and guide developers is an essential prerequisite to successfully adopt an ADR like this.

Reply:
I agree it may seem that they are too many rules, but I don't think they are that many in practice, just that the write-up feels like that.

Although, it may take you a few hours, or hopefully only a few days to get used to it, once you have the pattern established you don't need to learn new things.

The intention is to bring long term value since obvious there is some short term cost.